### ISSUE TITLE
Should `@Reducers` be decomposed into separate decorators?

### ISSUE THREAD

**Issue by KarlPurk** [KarlPurk/redux-decorators#11]
The more I look at the dual purpose of the `@Reducer` decorator the more I dislike it.  

It doesn't make sense to have one decorator serve 2 purposes.

There are a few options here.

1. Introduce a new `@RootReducer` decorator and move the root reducer functionality to this new decorator.
2. Introduce a new `@Action` decorator and move the action reducer functionality to this new decorator.

Perhaps a combination of the two?

Thoughts welcome.

**Comment by tdbrian** [comment]
I like the idea of an `@Action` decorator. Something like this?

``` javascript
@Action() increment(count) {
    return count + 1;
}
```

Is that what you're thinking about?

**Comment by KarlPurk** [comment]
Yes, pretty much.  It would also be used at the class-level too:

``` js
@Action('add', 'remove')
class MyReducers {
  add() {}
  remove() {}
}
```

We could have both plural and singular variations if required.  For me, it makes more sense to use the plural form when used against a class.

``` js
@Actions('add', 'remove')
class MyReducers {
  ...
}
```

But that doesn't feel right when used against a method:

``` js
@Actions() increment(count) {
    return count + 1;
}
```
